internal/ionoscloud/ionoscloud.go

```diff
@@ -0,0 +1,340 @@
+package ionoscloud
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ionos-cloud/external-dns-ionos-webhook/internal/ionos"
+	"github.com/ionos-cloud/external-dns-ionos-webhook/pkg/endpoint"
+	"github.com/ionos-cloud/external-dns-ionos-webhook/pkg/plan"
+	"github.com/ionos-cloud/external-dns-ionos-webhook/pkg/provider"
+	sdk "github.com/ionos-cloud/sdk-go-dns"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	logFieldZoneID        = "zoneID"
+	logFieldRecordID      = "recordID"
+	logFieldRecordName    = "recordName"
+	logFieldRecordType    = "recordType"
+	logFieldRecordContent = "recordContent"
+	logFieldRecordTTL     = "recordTTL"
+	logFieldDomainFilter  = "domainFilter"
+	// max number of records to read per request
+	recordReadLimit = 1000
+	// max number of records to read in total
+	recordReadMaxCount = 10 * recordReadLimit
+	// max number of zones to read per request
+	zoneReadLimit = 1000
+	// max number of zones to read in total
+	zoneReadMaxCount = 10 * zoneReadLimit
+)
+
+type DNSClient struct {
+	client *sdk.APIClient
+	dryRun bool
+}
+
+type DNSService interface {
+	GetAllRecords(ctx context.Context, offset int32) (sdk.RecordReadList, error)
+	GetRecordsByZoneIdAndName(ctx context.Context, zoneId, name string) (sdk.RecordReadList, error)
+	GetZones(ctx context.Context, offset int32) (sdk.ZoneReadList, error)
+	DeleteRecord(ctx context.Context, zoneId string, recordId string) error
+	CreateRecord(ctx context.Context, zoneId string, record sdk.RecordCreate) error
+}
+
+// GetAllRecords retrieve all records https://github.com/ionos-cloud/sdk-go-dns/blob/master/docs/api/RecordsApi.md#recordsget
+func (c *DNSClient) GetAllRecords(ctx context.Context, offset int32) (sdk.RecordReadList, error) {
+	log.Debugf("get all records with offset %d ...", offset)
+	records, _, err := c.client.RecordsApi.RecordsGet(ctx).Limit(recordReadLimit).Offset(offset).FilterState(sdk.AVAILABLE).Execute()
+	if err != nil {
```

Comment:
Nice to introduce a more advanced logging here - but this could be harmonized with the ionoscore implementations as well.

Reply:
sure, and probably this not the only topic. However, (at least) for this PR, is  would like leave the ionoscore code untouched ( more or less). Im also not sure about the release, but lets talk directly about that.

Reply:
As we discussed: we let the ionoscore implementation as it is for now.